hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/codec/TokenFreezeUnfreezeWrapper.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2020-2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.store.contracts.precompile.codec;
+
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.TokenID;
+
+public record TokenFreezeUnfreezeWrapper(TokenID token, AccountID account) {}
```

Comment:
Functions freeze, unfreeze, isFrozen receive the same tokenID and accountID arguments. Should there be a separate wrapper record for every function or is this approach with the single "TokenFreezeUnfreezeWrapper"  ok?

Reply:
We can have a common wrapper but separate building method for each precompile. This would make the code easier to maintain in the future. The methods can be:

- forFreeze
- forUnfreeze
- forIsFrozen

and they might reuse a common private method that initializes the wrapper.